addOns/graphql/src/main/java/org/zaproxy/addon/graphql/GraphQlFingerprinter.java

```diff
@@ -54,6 +61,7 @@ public class GraphQlFingerprinter {
     public GraphQlFingerprinter(URI endpointUrl) {
         requestor = new Requestor(endpointUrl, HttpSender.MANUAL_REQUEST_INITIATOR);
         queryCache = new HashMap<>();
+        addEngineHandler(this::raiseFingerprintingAlert);
```

Comment:
I don't think this is better than just call the method, it's adding to a class variable and never removing them, which means several alerts will end up being raised for different endpoints and will be leaking instances (at least until the tech detect is uninstalled but will leak again).

Reply:
So just don't add it internally call it when needed but leave the impl as-is?

Reply:
Yes.